estimage/simpledata.py

```
@@ -35,24 +35,48 @@ def CONFIG_FILENAME(cls):
+        self.backend = backend
 
+    def _get_io(self, of_what, category, datadir=None):
+        ret = persistence.get_persistence(of_what, self.backend)
+        path = self._get_filepath(category, datadir)
+        self._set_file_path(ret, path)
 
-class ProjCardIO(IniInDirMixin, ini.IniCardIO):
-    CONFIG_BASENAME = "projective.ini"
-    WHAT_IS_THIS = "projective card"
+    def _get_filepath(self, category):
+        if not datadir:
```

Ruff (0.13.1)

Undefined name `datadir`

(F821)